app/actions/deposit.ts

```diff
@@ -0,0 +1,98 @@
+import IERC20 from "@klimadao/lib/abi/IERC20.json";
+import ToucanPoolToken from "@klimadao/lib/abi/ToucanPoolToken.json";
+import { addresses } from "@klimadao/lib/constants";
+import { formatUnits as formatUnitsFn } from "@klimadao/lib/utils";
+import { BigNumber, Contract, providers } from "ethers";
+import { formatUnits, parseUnits } from "ethers-v6";
+import { CarbonToken } from "lib/queryUserCarbonTokens";
+import { OnStatusHandler } from "./utils";
+
+export const getAllowance = async (params: {
+  provider: providers.JsonRpcProvider;
+  selectedToken: Pick<CarbonToken, "amount" | "token">;
+  address: string;
+}) => {
+  try {
+    const contract = new Contract(
+      params.selectedToken.token.id,
+      IERC20.abi,
+      params.provider.getSigner()
+    );
+    const value: BigNumber = await contract.allowance(
+      params.address,
+      addresses["mainnet"]["bct"]
```

Comment:
Can "mainnet" & "bct" be parametrised?

Reply:
Was keeping with the general convention already used across the dApp. Although it should actually be `addresses["mainnet"].bct` to keep with the current naming convention.